Yank/repex.py

```diff
@@ -2089,31 +2118,29 @@ def _restore_dict_from_netcdf(self, ncgrp):
         options = dict()
 
         import numpy
+        from .utils import quantity_from_string
         for option_name in ncgrp.variables.keys():
             # Get NetCDF variable.
             option_ncvar = ncgrp.variables[option_name]
             type_name = getattr(option_ncvar, 'type')
+            # TODO: Remove the if/elseif structure into one handy function
             # Get option value.
             if type_name == 'NoneType':
                 option_value = None
-            elif option_ncvar.shape == ():
-                option_value = option_ncvar.getValue()
-                # Cast to python types.
-                if type_name == 'bool':
-                    option_value = bool(option_value)
-                elif type_name == 'int':
-                    option_value = int(option_value)
-                elif type_name == 'float':
-                    option_value = float(option_value)
-                elif type_name == 'str':
-                    option_value = str(option_value)
-            elif (option_ncvar.shape[0] >= 0):
-                option_value = np.array(option_ncvar[:], eval(type_name))
-                # TODO: Deal with values that are actually scalar constants.
-                # TODO: Cast to appropriate type
-            else:
-                option_value = option_ncvar[0]
-                option_value = eval(type_name + '(' + repr(option_value) + ')')
+            else: # Handle all Types not None
+                option_type = self._convert_netcdf_store_type(type_name)
+                if option_ncvar.shape == ():
+                    # Handle Standard Types
+                    option_value = option_type(option_ncvar.getValue())
+                elif (option_ncvar.shape[0] >= 0):
+                    # Handle array types
+                    option_value = np.array(option_ncvar[:], option_type)
+                    # TODO: Deal with values that are actually scalar constants.
+                    # TODO: Cast to appropriate type
+                else:
+                    # Handle iterable types?
+                    # TODO: Figure out what is actually cast here
+                    option_value = option_type(option_ncvar[0])
```

Comment:
Changes made to ncvar type inputs have been mostly rolled into the function `_convert_netcdf_store_type`. I just want to make sure there is not a type handled by the old `elif` format that would be handled unexpectedly from the new function.